catchup/universalFetcher.go

```diff
@@ -90,21 +90,21 @@ func (uf *universalBlockFetcher) fetchBlock(ctx context.Context, round basics.Ro
 	return block, cert, downloadDuration, err
 }
 
-func processBlockBytes(fetchedBuf []byte, r basics.Round, debugStr string) (blk *bookkeeping.Block, cert *agreement.Certificate, err error) {
+func processBlockBytes(fetchedBuf []byte, r basics.Round, peerAddr string) (blk *bookkeeping.Block, cert *agreement.Certificate, err error) {
 	var decodedEntry rpcs.EncodedBlockCert
 	err = protocol.Decode(fetchedBuf, &decodedEntry)
 	if err != nil {
-		err = fmt.Errorf("fetchBlock(%d): cannot decode block from peer %v: %v", r, debugStr, err)
+		err = makeCannotDecodeBlockError(r, peerAddr, err.Error())
```

Comment:
That would defeat the purpose of having a customized error object. Instead, pass the err to the `makeCannotDecodeBlockError` method. The message formatting can remain the same, but you should add an Unwrap method which would return the inner error. Read https://blog.golang.org/go1.13-errors for more info.

Reply:
I'll add the Unwrap method. However, in this case it is not useful. msgp returns msgp.errWrapped, which does not implement Unwrap, and it is only used with Error() call, which msgp internally unwraps.